pkg/webhooks/pod_admission_webhook.go

```diff
@@ -99,7 +99,7 @@ func (p *podWebhook) leaderPodForFollower(ctx context.Context, pod *corev1.Pod)
 
 	// Validate there is only 1 leader pod for this job.
 	if len(podList.Items) != 1 {
-		return nil, fmt.Errorf("incorrect number of leader pods for this job (expected 1, got %d)", len(podList.Items))
+		return nil, fmt.Errorf("leader pod (%s) does not exist or has not been scheduled yet (expected 1 pod, got %d)", leaderPodName, len(podList.Items))
```

Comment:
We wouldn't know here that it has not been scheduled, it will be above (the call to `leaderPodScheduled`); also leader pod doesn't exist is applicable only if the len is 0

Reply:
sorry, it's late and my brain is getting tired 😅 

updated this error message, and added a log line in the leaderPodScheduled function.